tika-parsers/src/main/java/org/apache/tika/parser/crypto/TSDParser.java

```
@@ -0,0 +1,333 @@
+            metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_DATE, tsdm.getEmitDateStr() + " " + this.TSD_DATE_FORMAT);
+            metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_POLICY_ID, tsdm.getPolicyId());
+            metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_SERIAL_NUMBER, tsdm.getSerialNumberFormatted());
+            metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_TSA, tsdm.getTSAstr());
+            metadata.set(TSD_LOOP_LABEL + count + " - " + this.TSD_ALGORITHM, tsdm.getAlgorithmName());
+            count++;
+        }
+                
+        XHTMLContentHandler xhtml = new XHTMLContentHandler(handler, metadata);
+        xhtml.startDocument();
+        xhtml.endDocument();
+        
+     }
+    
+    private List<TSDMetas> buildMetas(InputStream stream) {
+        
+        List<TSDMetas> tsdMetasList = new ArrayList<TSDMetas>();
+        
+        try {
+             
+             CMSTimeStampedDataParser cmsTimeStampedData = new CMSTimeStampedDataParser(stream);
+             
+             TimeStampToken[] tokens = cmsTimeStampedData.getTimeStampTokens();
+             
+             for (int i=0; i < tokens.length; i++) {
+                 
+                 TSDMetas tsdMetas = new TSDMetas(true,
+                                                   tokens[i].getTimeStampInfo().getGenTime(),
+                                                   tokens[i].getTimeStampInfo().getPolicy().getId(),
+                                                   tokens[i].getTimeStampInfo().getSerialNumber(),
+                                                   tokens[i].getTimeStampInfo().getTsa(),
+                                                   tokens[i].getTimeStampInfo().getHashAlgorithm().getAlgorithm().getId());
+                 
+                 tsdMetasList.add(tsdMetas);
+             }
+             
+        } catch (Exception ex) {
```

mb add logging instead of swallowing exception silently